pdata/pprofile/mappings_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofile
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetMapping(t *testing.T) {
+	table := NewMappingSlice()
+	m := NewMapping()
+	m.SetMemoryLimit(1)
+	m2 := NewMapping()
+	m2.SetMemoryLimit(2)
+	loc := NewLocation()
+
+	// Put a first mapping
+	require.NoError(t, SetMapping(table, loc, m))
+	assert.Equal(t, 1, table.Len())
+	assert.Equal(t, int32(0), loc.MappingIndex())
+
+	// Put the same mapping
+	// This should be a no-op.
+	require.NoError(t, SetMapping(table, loc, m))
+	assert.Equal(t, 1, table.Len())
+	assert.Equal(t, int32(0), loc.MappingIndex())
+
+	// Set a new mapping
+	// This adds an index and adds to the table.
+	require.NoError(t, SetMapping(table, loc, m2))
+	assert.Equal(t, 2, table.Len())
+	assert.Equal(t, int32(1), loc.MappingIndex())
+
+	// Set an existing mapping
+	require.NoError(t, SetMapping(table, loc, m))
+	assert.Equal(t, 2, table.Len())
+	assert.Equal(t, int32(0), loc.MappingIndex())
+}
```

Comment:
Just for completeness, especially since the `0` in the previous test is a default value.
```suggestion
	// Set another existing mapping
	require.NoError(t, SetMapping(table, loc, m2))
	assert.Equal(t, 2, table.Len())
	assert.Equal(t, int32(table.Len() - 1), loc.MappingIndex())
}
```